### PR TITLE
one vs. one logreg update

### PR DESCRIPTION
**Pull request by yugeji**

### CODE REVIEW

**Comment by falexwolf** [comment]
Nice, thank you! :smile:

File: scanpy/tools/rank_genes_groups.py [hunk 1]
```diff
@@ -181,19 +181,36 @@ def rank_genes_groups(
...
         from sklearn.linear_model import LogisticRegression
-        if reference != 'rest':
-            raise ValueError('\'logreg\' is only implemented for `reference==\'rest\'`.')
+        if reference not in groups:
+            raise Exception("reference must be in groups")
+        if len(groups) == 1:
+            raise Exception("Cannot perform logistic regression on a single cluster.")
+        if len(groups) == 2: #binary logistic regression
+            adata_copy = adata[(adata.obs[groupby] == groups_order[0]) | (adata.obs[groupby] == groups_order[1])]
+            X = adata_copy.raw.X
```
Comment (gokceneraslan) [inline scanpy/tools/rank_genes_groups.py:193]: What if use_raw is False?

Comment (falexwolf) [reply scanpy/tools/rank_genes_groups.py:193]: very right... i'll go over and correct it...

Comment (yugeji) [reply scanpy/tools/rank_genes_groups.py:193]: If you haven't yet, I can go fix it... sorry about that

**Pull request merged**